components/active/quiz/Survey.vue

```diff
@@ -0,0 +1,173 @@
+        >
+          <div class="flex h-10 w-10 items-center justify-center">
+            <AtomsIcon
+              v-if="isSelected(answer)"
+              name="carbon:radio-button-checked"
+              class="text-green-600"
+            />
+
+            <AtomsIcon
+              v-else
+              name="carbon:radio-button"
+              class="text-gray-400"
+            />
+          </div>
+
+          <input
+            :id="'answer' + answer"
+            :value="answer"
+            :name="currentQuestion"
+            :checked="isSelected(answer)"
+            type="radio"
+            class="hidden"
+            @input="() => select(answer)"
+          />
+          <span class="text-bold w-full py-4 text-sm">
+            {{ answer }}
+          </span>
+        </label>
+
+        <!-- Buttons -->
+        <div
+          :class="isFirstStep ? 'justify-end' : 'justify-between'"
+          class="mt-6 flex w-full flex-wrap items-center"
+        >
+          <AtomsButton v-if="!isFirstStep" color="gray" @click="goBack">
+            <AtomsIcon
+              name="carbon:arrow-left"
+              class="mr-2 text-lg text-gray-800"
+            />
+            {{ t('page.quiz.questions.back') }}
+          </AtomsButton>
+
+          <AtomsButton :disabled="!currentAnswer" color="gray" @click="goNext">
+            {{ t('page.quiz.questions.next') }}
+            <AtomsIcon
+              name="carbon:arrow-right"
+              class="ml-2 text-lg text-gray-800"
+            />
+          </AtomsButton>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+  import { useI18n } from 'vue-i18n';
+  import type { ApiQuizQuestions } from '~/types/api/quiz';
+
+  const props = defineProps<{
+    quiz: ApiQuizQuestions;
+  }>();
+
+  const emit = defineEmits<{
+    (e: 'submit', answers: string[]): void;
+  }>();
+
+  definePageMeta({
+    documentDriven: false,
+  });
+
+  const { t } = useI18n();
+
+  // Refs
+  const selectedAnswers = ref<string[]>([]);
+  const currentStepNumber = ref<number>(1);
+
+  // Computed
+  const max = computed<number>(() => {
+    return props.quiz.questions?.length;
+  });
+
+  const progress = computed(() => {
+    return Math.round((currentStepNumber.value / max.value) * 100);
+  });
+
+  const currentArrayIndex = computed(() => {
+    return currentStepNumber.value - 1;
+  });
+
+  const currentQuestion = computed(() => {
+    return props.quiz.questions?.[currentArrayIndex.value];
+  });
+
+  const currentAnswer = computed(() => {
+    return selectedAnswers.value[currentArrayIndex.value];
+  });
+
+  const isFirstStep = computed(() => {
+    return currentStepNumber.value === 1;
+  });
+
+  const isLastStep = computed(() => {
+    return currentStepNumber.value === max.value;
+  });
+
+  // Methods
+  function goBack() {
```

Comment:
isn't it better to use nested route? Your method will not work well with back/forward buttons operating on `window.history` object

Reply:
Doing it this way will add complexity. I can implement it if you insist on having it this way, but in my opinion, this will have little benefit.